client/src/main/proto/spine/validation.proto

```diff
@@ -61,19 +61,23 @@ extend google.protobuf.FieldOptions {
 
 // A validation constraint violation for a message field.
 message ConstraintViolation {
-    // An interpolated error message for this constraint violation.
+    // An error message for this constraint violation
+    // (or a template which contains `%s` format specifiers).
     string message = 1;
 
-    // A non-interpolated error message for this constraint violation.
-    string message_template = 2;
+    // Parameters (as strings) referenced by the format specifiers in the error message format string.
+    repeated string format_param = 2;
```

Comment:
Name it just `param`. Having `format` for the string part of the story would be enough.